app/components/new-project-modal/component.js

```diff
@@ -0,0 +1,77 @@
+import { computed } from '@ember/object';
+import { A } from '@ember/array';
+import { inject as service } from '@ember/service';
+import Component from '@ember/component';
+import { task, timeout } from 'ember-concurrency';
+
+export default Component.extend({
+    store: service(),
+    newNode: null,
+    more: false,
+    institutionsSelected: computed.oneWay('user.institutions'),
+
+    actions: {
+        select(institution) {
+            this.set('institutionsSelected', this.get('institutionsSelected').slice());
+            if (this.get('institutionsSelected').includes(institution)) {
+                this.get('institutionsSelected').removeObject(institution);
+            } else {
+                this.get('institutionsSelected').pushObject(institution);
+            }
+        },
+        selectAll() {
+            this.set('institutionsSelected', this.get('user.institutions').slice());
+        },
+        removeAll() {
+            this.set('institutionsSelected', A([]));
+        },
+        toggleMore() {
+            this.toggleProperty('more');
+        },
+        closeModal() {
+            this.set('newNode', null);
+            this.get('closeModal')();
+        },
+        reload() {
+            this.set('newNode', null);
+            this.get('closeModal')();
+            this.get('reloadNodes')();
+        },
+        pushEnter() {
+            if (this.get('nodeTitle.length')) {
+                this.get('createNode').perform();
+            }
+        },
+    },
+
+    modalClass: computed('styleNamespace', function() {
+        return `${this.get('styleNamespace')}__modal`;
+    }),
+
+    findNodes: task(function* (term) {
```

Comment:
I would rather not use generator functions for asynchronous flow control. This should be done with async/await--especially since Ember is build on top of promises. Generators are for a different purpose--namely reentering a function later (see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*))

I understand that this is a pattern from ember-concurrency, however, that seems outdated now. I would at least like to debate the merits of ember-concurrency before we decide that it is something we want to follow. IMO, less dependencies are usually the way to go and the web has largely settled on async/await.

Reply:
Ok for now, but we will discuss the merits of ember-concurrency.